File: src/main/java/io/bindernews/thegrackle/power/CoolingPhoenixPower.java
package io.bindernews.thegrackle.power;

import com.megacrit.cardcrawl.actions.common.ReducePowerAction;
import com.megacrit.cardcrawl.actions.watcher.ChangeStanceAction;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.powers.AbstractPower;
import com.megacrit.cardcrawl.stances.AbstractStance;
import io.bindernews.thegrackle.GrackleMod;
import io.bindernews.thegrackle.stance.StancePhoenix;

public class CoolingPhoenixPower extends BasePower {

    public static final String POWER_ID = GrackleMod.makeId(CoolingPhoenixPower.class);

    public CoolingPhoenixPower(AbstractCreature owner, int amount) {
        super(POWER_ID);
        setOwnerAmount(owner, amount);
        this.type = PowerType.DEBUFF;
        this.isTurnBased = true;
        loadRegion("frail");
        this.updateDescription();
    }

    @Override
    public void onChangeStance(AbstractStance oldStance, AbstractStance newStance) {
        if (StancePhoenix.isStance(newStance)) {
            addToTop(new ChangeStanceAction(oldStance));
        }
    }

    @Override
    public void atEndOfRound() {
        addToBot(new ReducePowerAction(owner, owner, POWER_ID, 1));
    }

    @Override
    public void updateDescription() {
        description = strings.DESCRIPTIONS[0];
    }

    @Override
    public AbstractPower makeCopy() {
        return new CoolingPhoenixPower(owner, amount);
    }
}

File: src/main/kotlin/net/bindernews/grackle/power/CoolingPhoenixPower.kt
package net.bindernews.grackle.power

import com.megacrit.cardcrawl.actions.common.ReducePowerAction
import com.megacrit.cardcrawl.actions.watcher.ChangeStanceAction
import com.megacrit.cardcrawl.core.AbstractCreature
import com.megacrit.cardcrawl.powers.AbstractPower
import com.megacrit.cardcrawl.stances.AbstractStance
import net.bindernews.grackle.GrackleMod
import net.bindernews.grackle.stance.StancePhoenix.Companion.isStance

class CoolingPhoenixPower(owner: AbstractCreature, amount: Int) : BasePower(POWER_ID) {
    init {
        setOwnerAmount(owner, amount)
        type = PowerType.DEBUFF
        isTurnBased = true
        loadRegion("frail")
        updateDescription()
    }

    override fun onChangeStance(oldStance: AbstractStance, newStance: AbstractStance) {
        if (isStance(newStance)) {
            addToTop(ChangeStanceAction(oldStance))
        }
    }

    override fun atEndOfRound() {
        addToBot(ReducePowerAction(owner, owner, POWER_ID, 1))
    }

    override fun updateDescription() {
        description = strings.DESCRIPTIONS[0]
    }

    override fun makeCopy(): AbstractPower {
        return CoolingPhoenixPower(owner, amount)
    }

    companion object {
        @JvmField val POWER_ID = GrackleMod.makeId(CoolingPhoenixPower::class.java)
    }
}